File: src/main/java/akko/ddbot/task/LiveRoomTask.java
package akko.ddbot.task;

import java.io.IOException;
import java.io.StringReader;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.List;

import akko.ddbot.BotMainActivity;
import akko.ddbot.InitCheck;
import akko.ddbot.data.BilibiliApi.BilibiliDataClass;
import akko.ddbot.data.BilibiliApi.Data;
import akko.ddbot.data.BilibiliApi.Live_room;
import akko.ddbot.sql.SQLFun;
import com.fasterxml.jackson.databind.ObjectMapper;

import akko.ddbot.network.BilibiliApiService;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Retrofit;


public class LiveRoomTask implements Runnable {

    @Override
    public void run() {
        ObjectMapper oMapper = new ObjectMapper();
        Retrofit retrofit = new Retrofit.Builder().baseUrl("https://api.bilibili.com/x/space/acc/").build();
        try {
            while (true)
            {
                new LiverInit();
                List<String> liverList = LiverInit.liverList;
                Connection sqliteC = new SQLFun().connection("GroupInfo");
                for (String vID : liverList) {
                    Call<ResponseBody> call = retrofit.create(BilibiliApiService.class).getDatCall(vID);
                    String body = null;
                    ResponseBody rawBody = call.execute().body();
                    if (rawBody != null) { body = rawBody.string(); }
                    else {
                        BotMainActivity.bot.getAccountManager().getNonAccountSpecifiedApi().sendGroupMsg(InitCheck.GROUP_ID,"b站网络出问题了(确信");
                        continue;
                    }
                    Data data = oMapper.readValue(body, BilibiliDataClass.class).getData();
                    Live_room liveRoomData = data.getLive_room();
                    int statusRightNow = liveRoomData.getLiveStatus();
                    System.out.println(vID);
                    int statusindb = sqliteC.prepareStatement("select * from  vLiver WHERE vID = '" + vID + "';").executeQuery().getInt("vSTATE");

                    if (statusRightNow == 1 && statusindb == 0)
                    {
                        sqliteC.prepareStatement("UPDATE vLiver SET vSTATE = 1 WHERE vID = " + vID + ";").execute();
                        new RemindListener().RemindListenerFun(liveRoomData.getCover(),vID,data.getName(),liveRoomData.getTitle(),liveRoomData.getUrl());;
                    }
                    else if (statusRightNow == 0 && statusindb == 1)
                    {
                        sqliteC.prepareStatement("UPDATE vLiver SET vSTATE = 0 WHERE vID = " + vID + ";").execute();
                    }
                }
                sqliteC.close();
                Thread.sleep(60000);
            }
        } 
        catch (SQLException | InterruptedException | IOException e)
        {
            System.out.println(e.toString());
            e.printStackTrace();
        }
    }
    
}

File: src/main/java/akko/ddbot/task/LiveRoomTask.kt
package akko.ddbot.task

import akko.ddbot.BotMainActivity
import akko.ddbot.Init
import akko.ddbot.data.BilibiliApi.BilibiliDataClass
import akko.ddbot.network.BilibiliApiService
import akko.ddbot.sql.KtormObject
import akko.ddbot.sql.connectionPool
import akko.ddbot.utilities.groupMsg
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import okhttp3.ResponseBody
import org.ktorm.database.Database
import org.ktorm.dsl.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import java.io.IOException
import java.sql.SQLException

val LiveRoomTask = Thread {
    val oMapper = jacksonObjectMapper()
    val retrofit = Retrofit.Builder().baseUrl("https://api.bilibili.com/x/space/acc/").build()
    try {
        while (true) {
            LoadLiver.reLoad()
            val liverList: List<String> = LoadLiver.liverList
            for (vID in liverList) {
                val call = retrofit.create(BilibiliApiService::class.java).getDatCall(vID)
                call!!.enqueue(object: Callback<ResponseBody?>{
                    override fun onResponse(call: Call<ResponseBody?>, response: Response<ResponseBody?>) {
                        if (response.body() != null) {
                            val database = Database.connect(connectionPool.connectionPool)
                            val body = response.body()!!.string()
                            val data = oMapper.readValue<BilibiliDataClass>(body).data
                            val liveRoomData = data.live_room
                            val statusRightNow = liveRoomData.liveStatus
                            BotMainActivity.NLogger!!.log("$vID 检查完成")
                            var statusindb = 0
                            for (row in database.from(KtormObject.VliverInfo).select(KtormObject.VliverInfo.liveState).where {KtormObject.VliverInfo.pid like vID})
                            {
                                statusindb = row[KtormObject.VliverInfo.liveState]!!
                            }
                            if (statusRightNow == 1 && statusindb == 0) {
                                database.update(KtormObject.VliverInfo)
                                {
                                    set(it.liveState,1)
                                    where { it.pid like vID }
                                }
                                remindListenerFun(liveRoomData.cover, vID, data.name, liveRoomData.title, liveRoomData.url)
                            } else if (statusRightNow == 0 && statusindb == 1) {
                                database.update(KtormObject.VliverInfo)
                                {
                                    set(it.liveState,0)
                                    where { it.pid like vID }
                                }
                            }
                        } else {
                            groupMsg(Init.GROUP_ID.toLong(), "b站网络出问题了(确信")
                        }
                    }
                    override fun onFailure(call: Call<ResponseBody?>, t: Throwable) {
                        t.printStackTrace()
                    }
                })
            }
            Thread.sleep(60000)
        }
    } catch (e: SQLException) {
        BotMainActivity.ExceptionLogger!!.debug(e.message)
        e.printStackTrace()
    } catch (e: InterruptedException) {
        BotMainActivity.ExceptionLogger!!.debug(e.message)
        e.printStackTrace()
    } catch (e: IOException) {
        BotMainActivity.ExceptionLogger!!.debug(e.message)
        e.printStackTrace()
    }
}